modules/claims_api/lib/custom_error.rb

```diff
@@ -8,84 +8,29 @@ def initialize(error, claim, method)
       @method = method
     end
 
-    def build_error # rubocop:disable Metrics/MethodLength
-      get_status
-      get_code
-      get_source
-      if @error.is_a?(Faraday::ConnectionFailed) || @error.is_a?(Faraday::ParsingError) ||
-         @error.is_a?(Faraday::NilStatusError) || @error.is_a?(Faraday::TimeoutError) ||
-         @error.is_a?(::Common::Exceptions::BackendServiceException) ||
-         @error.is_a?(::Common::Exceptions::ExternalServerInternalServerError) ||
-         @error.is_a?(::Common::Exceptions::BadGateway) || @error.is_a?(Faraday::SSLError) ||
-         @error.is_a?(Faraday::ServerError)
-        errors = { errors: [{ 'title' => 'Service Exception',
-                              'key' => @source,
-                              'detail' => 'A re-tryable error has occurred, original_error: ' \
-                                          "#{@error}.", status: @status, code: @code }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::ServiceError, errors
-
-      elsif @error.is_a?(StandardError) || @error.is_a?(Faraday::BadRequestError) ||
-            @error.is_a?(Faraday::ConflictError) || @error.is_a?(Faraday::ForbiddenError) ||
-            @error.is_a?(Faraday::ProxyAuthError) || @error.is_a?(Faraday::ResourceNotFound) ||
-            @error.is_a?(Faraday::UnauthorizedError) || @error.is_a?(Faraday::UnprocessableEntityError) ||
-            @error.is_a?(Faraday::ClientError)
-
-        errors = { errors: [{ 'title' => 'Client error',
-                              'key' => @source,
-                              'detail' => 'A client exception has occurred, job will not be re-tried. ' \
-                                          "original_error: #{@error}.", status: '400', code: '400' }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::BadRequest, errors
+    def handle_errors
+      case @error
+      when Faraday::ParsingError
+        raise_backend_exception('EVSS502', self.class)
+      when ::Common::Exceptions::BackendServiceException
+        raise ::Common::Exceptions::Forbidden if @error&.original_status == 403
+
+        raise_backend_exception('EVSS400', self.class, @error) if @error&.original_status == 400
+        raise ::Common::Exceptions::Unauthorized if @error&.original_status == 401
       else
-        errors = { errors: [{ 'title' => 'Unknown error',
-                              'key' => @source,
-                              'detail' => 'An unknown error has occurred, and the custom_error file may ' \
-                                          "need to be modified. original_error: #{@error}.", status: @status,
-                              code: @code }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::ServiceError, errors
+        raise @error
       end
     end
```

Comment:
@stiehlrod  I'm not sure this is ever going to show a source value when viewing the claim.  It looks like for every `::Common::Exceptions::BackendServiceException` we will be passing in `EVSS***` as the key.  The claims_controller did use that key to show the `source` though.


<img width="1060" alt="Screenshot 2024-03-21 at 10 08 29 AM" src="https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/52112dba-0502-4b01-ad46-460f999b118d">


![Screenshot 2024-03-21 at 10 08 51 AM](https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/aa662d1c-c431-4ad9-b562-37de453fb829)

Reply:
Added back the `get_source` method

Reply:
get source only adds to what we save in the DB... The source in the above pulls from the `key` field, so it looks like the below.

```
"source": "header.va_eauth_birlsfilenumber.Invalid"
```

not sure I like pointing to where it threw the error in the code base. I'd almost rather keep `self.class` but I'd even prefer over both `@error.class`. 

```
(byebug) @source
"vets-api/lib/common/client/middleware/response/raise_error.rb"
(byebug) self.class
ClaimsApi::CustomError
(byebug) @error.class
Common::Exceptions::BackendServiceException
```

Reply:
My concern is that the old return would show this as something like `form526/submit/clientError` (it goes in as `form526.submit.clinetError`).  So is changing this in the response something we can do with v1?  If it is then my question at least is probably a moot point.

Reply:
didn't see this last bit. we're not changing the response structure, just the literal string internal of it, so it's not a breaking change.. but it is worth considering. if we previously returned something similar to the key, we could do that here. most of those came from the `key` field if I recall.... which the `validate` endpoint uses
```
            { status: 422, detail: "#{error['key']} #{error['detail']}", source: error['key'] }

```

the v1 claims controller does similar.
```
        errors.map do |error|
          formatted = error['key'] ? error['key'].gsub('.', '/') : error['key']
          { status: 422, detail: "#{error['severity']} #{error['detail'] || error['text']}".squish, source: formatted }
        end
```

Reply:
@FonzMP & @rockwellwindsor-va is this what you are proposing:
```
    def raise_backend_exception(key, source, error = nil)
      formatted_key = @error.key ? @error.key.gsub('.', '/') : @error.key
      raise ::Common::Exceptions::BackendServiceException.new(
        key,
        { source: formatted_key },
        error&.original_status,
        error&.original_body
      )
    end
```
?

Reply:
@stiehlrod I _believe_ `, source: formatted }`  is the value being discussed in the comment, which comes from the `key:` in the error message.  So looking at the error above, there does not appear to be a `key:` key, so the `source` would not get populated in the evss formatter in the controllers.  I _believe_ the change here would be to have something like `{ key: formatted_key },` in the `raise_backend_exception` method so that when the error message gets to the formatters in the controllers it can grab the `error[:key]` value and put it into `, source: formatted }` in the message.  In that returned message source looks like `forms526/this/is/the/path` so it may be just switching the current `{ source: formatted_key },` to `{ key: formatted_key },` gets it in there